apps/services/auth/delegation-api/src/app/delegations/delegation-index.controller.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Headers,
+  Put,
+  UseGuards,
+} from '@nestjs/common'
+import * as kennitala from 'kennitala'
+import {
+  DelegationRecordDTO,
+  DelegationsIndexService,
+  CreateDelegationRecordInputDTO,
+} from '@island.is/auth-api-lib'
+import { AuthDelegationType } from '@island.is/shared/types'
+import { Documentation } from '@island.is/nest/swagger'
+import {
+  Auth,
+  CurrentAuth,
+  IdsAuthGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import { AuthScope } from '@island.is/auth/scopes'
+import { ApiSecurity } from '@nestjs/swagger'
+import { Audit, AuditService } from '@island.is/nest/audit'
+
+const namespace = '@island.is/auth/delegation-api/delegation-index'
+
+const parseDelegationInfo = (delegationInfo: string) => {
+  const [type, toNationalId, fromNationalId] = delegationInfo.split('_')
+
+  if (!type || !toNationalId || !fromNationalId) {
+    throw new BadRequestException('Invalid delegation information')
+  }
+
+  if (!kennitala.isValid(toNationalId) || !kennitala.isValid(fromNationalId)) {
+    throw new BadRequestException('Invalid national id')
+  }
+
+  return {
+    type: type as AuthDelegationType,
+    toNationalId,
+    fromNationalId,
+  }
+}
+
+const requestDocumentation = {
+  header: {
+    'X-Param-Id': {
+      required: true,
+      description:
+        'Delegation information delimited by an underscore e.g. delegationType_nationalIdTo_nationalIdFrom',
+    },
+  },
+}
+
+@UseGuards(IdsAuthGuard, ScopesGuard)
+@Scopes(AuthScope.delegationIndexWrite)
+@ApiSecurity('ias', [AuthScope.delegationIndexWrite])
+@Controller({
+  path: 'delegation-index',
+})
+@Audit({ namespace })
+export class DelegationIndexController {
+  constructor(
+    private readonly delegationIndexService: DelegationsIndexService,
+    private readonly auditService: AuditService,
+  ) {}
+
+  @Put('.id')
+  @Documentation({
+    description: 'Create or update a delegation index record.',
+    response: { status: 200, type: DelegationRecordDTO },
+    request: requestDocumentation,
+  })
+  async createOrUpdateDelegationRecord(
+    @CurrentAuth() auth: Auth,
+    @Headers('X-Param-Id') delegationInfo: string,
+    @Body() body: CreateDelegationRecordInputDTO,
+  ) {
```

Comment:
Add explicit return type for readability 
```suggestion
  ): Promise<DelegationRecordDTO> {
```